packages/services/cluster/src/types.rs

```diff
@@ -190,3 +191,48 @@ pub enum TlsState {
 	Active = 1,
 	Renewing = 2,
 }
+
+/// See `rivet_config::config::server::rivet::GuardPublicHostname` for docs.
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Hash)]
+pub enum GuardPublicHostname {
+	DnsParent(String),
+	Static(String),
+}
+
+impl GuardPublicHostname {
+	pub fn from_columns(
+		config: &rivet_config::Config,
+		datacenter_id: Uuid,
+		gph_dns_parent: Option<String>,
+		gph_static: Option<String>,
+	) -> GlobalResult<GuardPublicHostname> {
+		let gph = match (gph_dns_parent, gph_static) {
+			(Some(x), None) => GuardPublicHostname::DnsParent(x),
+			(None, Some(x)) => GuardPublicHostname::Static(x),
+			(Some(_), Some(_)) => bail!(
+				"guard public hostname dns parent & static cannot be defined at the same time"
+			),
+			(None, None) => {
+				if let Ok(domain_job) = config.server()?.rivet.domain_job() {
+					// Fall back to auto-generated hostname
+					let hostname = format!("actor.{}.{domain_job}", datacenter_id);
+					crate::types::GuardPublicHostname::DnsParent(hostname)
+				} else {
+					tracing::warn!(?datacenter_id, "no guard public hostname specified");
+					crate::types::GuardPublicHostname::Static(
+						"no-guard-public-hostname-specified-in-config.invalid".into(),
+					)
```

Comment:
why do we handle the error like this instead of erroring? seems hacky

Reply:
was a chicken or egg with upgrading dc workflow. you had to list dc in order to be able to upgrade the dcs, but this would throw an error if the new value was not populated. no longer an issue, though.